src/pages/ReimbursementAccount/RequestorStep.js

```diff
@@ -8,40 +9,71 @@ import TextLink from '../../components/TextLink';
 import Navigation from '../../libs/Navigation/Navigation';
 import CheckboxWithLabel from '../../components/CheckboxWithLabel';
 import Text from '../../components/Text';
-import {goToWithdrawalAccountSetupStep} from '../../libs/actions/BankAccounts';
+import {goToWithdrawalAccountSetupStep, setupWithdrawalAccount} from '../../libs/actions/BankAccounts';
 import Button from '../../components/Button';
 import FixedFooter from '../../components/FixedFooter';
 import IdentityForm from './IdentityForm';
+import Growl from '../../libs/Growl';
 
 class RequestorStep extends React.Component {
     constructor(props) {
         super(props);
 
+        this.submit = this.submit.bind(this);
+
         this.state = {
-            firstName: '',
-            lastName: '',
-            street: '',
-            city: '',
-            state: '',
-            zipCode: '',
-            dob: '',
-            ssnLast4: '',
+            firstName: lodashGet(props, ['achData', 'firstName'], ''),
+            lastName: lodashGet(props, ['achData', 'lastName'], ''),
+            requestorAddressStreet: lodashGet(props, ['achData', 'requestorAddressStreet'], ''),
+            requestorAddressCity: lodashGet(props, ['achData', 'requestorAddressCity'], ''),
+            requestorAddressState: lodashGet(props, ['achData', 'requestorAddressState'], ''),
+            requestorAddressZipCode: lodashGet(props, ['achData', 'requestorAddressZipCode'], ''),
+            dob: lodashGet(props, ['achData', 'dob'], ''),
+            ssnLast4: lodashGet(props, ['achData', 'ssnLast4'], ''),
+            isControllingOfficer: lodashGet(props, ['achData', 'isControllingOfficer'], false),
         };
     }
 
+    onFieldChange(field, value) {
+        const renamedFields = {
+            street: 'requestorAddressStreet',
+            city: 'requestorAddressCity',
+            state: 'requestorAddressState',
+            zipCode: 'requestorAddressZipCode',
+        };
+        const fieldName = lodashGet(renamedFields, field, field);
+        this.setState({[fieldName]: value});
+    }
+
+    validate() {
+        if (!this.state.isControllingOfficer) {
+            Growl.show(this.props.translate('requestorStep.isControllingOfficerError'), CONST.GROWL.ERROR);
+            return false;
+        }
+
+        return true;
+    }
+
+    submit() {
+        if (!this.validate()) {
+            return;
+        }
+        setupWithdrawalAccount({...this.state});
```

Comment:
I'm not 100% sure how this works in Web-Secure. But it seems like _maybe_ we should have the Onfido `sdkToken` after calling this and then can simply redirect back to this page again:

**Web-Secure API logic** - https://github.com/Expensify/Web-Secure/blob/bdccc8e969e8d1c79069d8add1b658cbcf23a326/site/app/dialogs/reimbursementAccount/reimbursementAccountPage.jsx#L273-L283

**Expensify.cash API logic** - https://github.com/Expensify/Expensify.cash/blob/125dbf3c455b37913c702bb5eb5c792bc30080ed/src/libs/actions/BankAccounts.js#L537-L554

Then we can pass that token to `<Onfido />` like we are doing here

https://github.com/Expensify/Expensify.cash/blob/125dbf3c455b37913c702bb5eb5c792bc30080ed/src/pages/EnablePayments/OnfidoStep.js#L68-L69

and just handle the callback/error and we should be good ?

Reply:
Yep! Working on that now